### PR TITLE
feat(run-protocol): charge penalty for liquidation

### LINKED ISSUE #4775: [Vaults] When Vaults are liquidated, charge a fee
## What is the Problem Being Solved?

When vaults are liquidated, a fee should be charged and accrued to the reward pool

## Description of the Design

New governance parameter `liquidationPenalty`, defaulting to 10%. Govern per VaultManager.

Proceeds from the penalty will go to the reserve pool. For now hold them in a `penaltyPoolSeat`.

Eventually make a timer for batching emptying the penalty seat into the reserve. The reserve doesn’t benefit from getting it sooner.

## Security Considerations

It's governed, but otherwise not a security issue

## Test Plan

add verification in the existing liquidation tests.

### PR DESCRIPTION
**Pull request by turadg**
closes: #4775

## Description

- governance param LiquidationPenalty on the vault manager (so it’s per collateral type)
- defaults to 10%
- charging logic is in the Vault Factory’s liquidation helper so that the liquidation contract is clean and if governance changes the contract or strategy it can’t change the penalty behavior
- penalty proceeds stored in a penaltyPoolSeat on the VM
- not actually transferred until #5012 so for tests use a getPenaltyAllocation akin to getRewardAllocation

### Security Considerations

Creates a new holder of assets, the `penaltyPoolSeat`.

### Documentation Considerations

Backwards compatible, no breaking changes.

### Testing Considerations

New tests

### CODE REVIEW

File: packages/run-protocol/src/econ-behaviors.js [hunk 1]
```diff
@@ -27,6 +27,16 @@ const BASIS_POINTS = 10_000n;
...
 const CENTRAL_DENOM_NAME = 'urun';
 
+/**
+ * @file A collection of productions, each of which declares inputs and outputs.
+ * Each function is passed a set of powers for reading from and writing to the vat config.
+ *
+ * Each of the things they produce they're responsible for resolving or setting.
+ *
+ * In production called by @agoric/vats to bootstrap.
+ * In testing called by differenty.
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/econ-behaviors.js]: maybe drop this line?

File: packages/run-protocol/src/vaultFactory/liquidation.js [hunk 2]
```diff
@@ -3,11 +3,30 @@
...
 const trace = makeTracer('LIQ');
 
+/**
+ * @param {Amount<'nat'>} proceeds
+ * @param {Amount<'nat'>} debt - after incurring penalty
+ * @param {Amount<'nat'>} penaltyPortion
+ */
+const partitionProceeds = (proceeds, debt, penaltyPortion) => {
+  const isShortfall = !AmountMath.isGTE(proceeds, debt);
+  const debtPaid = isShortfall ? proceeds : debt;
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/vaultFactory/liquidation.js]:
```suggestion
  const debtPaid = AmountMath.min(proceeds, debt);
```

File: packages/run-protocol/src/vaultFactory/liquidation.js [hunk 3]
```diff
@@ -31,10 +52,16 @@ const liquidate = async (
...
   collateralBrand,
+  penaltyPoolSeat,
+  penaltyRate,
 ) => {
   innerVault.liquidating();
-  const debt = innerVault.getCurrentDebt();
-  const { brand: runBrand } = debt;
+
+  const debtBeforePenalty = innerVault.getCurrentDebt();
+  const penalty = floorMultiplyBy(debtBeforePenalty, penaltyRate);
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/vaultFactory/liquidation.js]: The contract usually gets the benefit of rounding:
```suggestion
  const penalty = ceilMultiplyBy(debtBeforePenalty, penaltyRate);
```

File: packages/run-protocol/src/reserve/assetReserve.js [hunk 4]
```diff
@@ -11,7 +11,7 @@ import { AMM_INSTANCE } from './params.js';
 const makeLiquidityKeyword = keyword => `${keyword}Liquidity`;
 
 /**
- * Collateral Reserve holds onto Collateral for the RUN protocol, and can
+ * Asset Reserve holds onto assets for the RUN protocol, and can
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/reserve/assetReserve.js]: renaming the collateralReserve seems unrelated to the point of this PR.

**Review by Chris-Hibbert — APPROVED** [review]
Looks good.

File: packages/run-protocol/src/reserve/assetReserve.js (hunk 4 above)
Comment (turadg) [reply packages/run-protocol/src/reserve/assetReserve.js]: moved to https://github.com/Agoric/agoric-sdk/pull/5028

File: packages/run-protocol/src/vaultFactory/liquidation.js (hunk 2 above)
Comment (turadg) [reply packages/run-protocol/src/vaultFactory/liquidation.js]: 👍 I suppose `isShortfall` was defined for tracing but I already took it out of the trace because `debtPaid` implies it.
